instrumentation/kafka-clients/src/main/java/brave/kafka/KafkaTracing.java

```diff
@@ -0,0 +1,58 @@
+package brave.kafka;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracing;
+import brave.propagation.TraceContext;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import zipkin.internal.Util;
+
+/**
+ * Use this class to decorate your Kafka consumer / producer and enable Tracing.
+ */
+public final class KafkaTracing {
+
+  public static KafkaTracing create(Tracing tracing) {
+    return new KafkaTracing(tracing);
+  }
+
+  final Tracing tracing;
+
+  KafkaTracing(Tracing tracing) { // hidden constructor
+    if (tracing == null) {
+      throw new NullPointerException("tracing == null");
+    }
+    this.tracing = tracing;
+  }
+
+  public <K, V> Consumer<K, V> consumer(Consumer<K, V> consumer) {
+    return new TracingConsumer<>(tracing, consumer);
+  }
+
+  public <K, V> Producer<K, V> producer(Producer<K, V> producer) {
+    return new TracingProducer<>(tracing, producer);
+  }
+
+  /**
+   * Continues the trace extracted from the headers or creates a new one if the extract fails. Call
+   * this method while consumming your kafka records.
+   */
+  public Span nextSpan(ConsumerRecord record) {

```

Comment:
Todo : 
Expose this as a java8 function.
`Function<ConsumerRecord, Span> nextSpanFromRecord();`